src/OmniSharp.LanguageServerProtocol/Helpers.cs

```diff
@@ -96,7 +97,41 @@ public static DiagnosticSeverity ToDiagnosticSeverity(string logLevel)
         }
 
         public static DocumentUri ToUri(string fileName) => DocumentUri.File(fileName);
-        public static string FromUri(DocumentUri uri) => uri.GetFileSystemPath().Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+
+        public static DocumentUri ToUri(MetadataSource mds) =>
+            new Uri($"osmd:/Project/{mds.ProjectName}/Assembly/{mds.AssemblyName}/Symbol/{mds.TypeName}.cs");
```

Comment:
`omnisharp://metadata/Project/{mds.ProjectName}/Assembly/{mds.AssemblyName}/Symbol/{mds.TypeName}.cs` (follows closer with `file://`)

Reply:
Done.

Also note that I kept a single slash after `omnisharp:` as URIs (as I understand) require two slashes if we provide "authority" (i.e. hostname for regular http URIs) as the first component on the path. In this case the authority is implicit, right? I can do both versions -- I just thought a single slash makes more sense.

Reply:
just as long as it works in vscode in the future, will it work? I honestly don't know.  If we run into problems we'll have to make sure we document the changes.